charts/super-agent-deployment/templates/_helpers.tpl

```diff
@@ -40,3 +40,26 @@ If you need a list of TODOs, just `grep TODO` on the `values.yaml` and look for
   {{- .Values.config.content | toYaml -}}
 {{- end -}}
 {{- end -}}
+
+{{- /* These are the defaults that are used for all the containers in this chart */ -}}
+{{- define "newrelic-super-agent.securityContext.containerDefaults" -}}
+runAsUser: 1000
+runAsGroup: 2000
+allowPrivilegeEscalation: false
+readOnlyRootFilesystem: true
+{{- end -}}
+
+{{- /* Allow to change pod defaults dynamically */ -}}
+{{- define "newrelic-super-agent.securityContext.container" -}}
+{{- $defaults := fromYaml ( include "newrelic-super-agent.securityContext.containerDefaults" . ) -}}
+{{- $commonLibrary := include "newrelic.common.securityContext.container" . | fromYaml -}}
+
+{{- $finalSecurityContext := dict -}}
+{{- if $commonLibrary -}}
+    {{- $finalSecurityContext = mustMergeOverwrite $commonLibrary  -}}
+{{- else -}}
+    {{- $finalSecurityContext = $defaults  -}}
+{{- end -}}
+
+{{- toYaml $finalSecurityContext -}}
```

Comment:
NIT:
```suggestion
{{- if $commonLibrary -}}
    {{- toYaml $commonLibrary  -}}
{{- else -}}
    {{- toYaml $defaults  -}}
{{- end -}}
```

Reply:
you are right! it looks way better as you suggested!